internal/concatjs/concatjs.go

```diff
@@ -27,10 +27,10 @@ import (
 //
 // Example usage:
 //   http.Handle("/app_combined.js",
-// 	     concatjs.ServeConcatenatedJS("my/app/web_srcs.MF", ".", nil))
+// 	     concatjs.ServeConcatenatedJS("my/app/web_srcs.MF", ".", [], [], nil))
 //
 // Relative paths in the manifest are resolved relative to the path given as root.
-func ServeConcatenatedJS(manifestPath, root string, fs FileSystem) http.Handler {
+func ServeConcatenatedJS(manifestPath string, root string, preScripts []string, postScripts []string, fs FileSystem) http.Handler {
```

Comment:
Do you need both per and post scripts?

In the internal design, we just have `concatjs` pick up the `IBAZEL_...` environment variable. That's a bit more magical, but OTOH avoids having an interface dependency, and avoids having whoever calls `ServeConcatenatedJS` deal with the livereload bits.

Maybe that's easier?

Reply:
It could. I thought it was odd to have a concatenation aware of scripts. This way it stays generic. Also the pre and post scripts would be needed either way in the current design as requirejs is included as a preScript and the postScript is used for a the snipped `"require([\"%s\"]);", *entryModule` to start the application code.

Reply:
Right. You could move those to the bazel rules generating this if you wanted to to avoid an API dependency on that, which might give a cleaner interface. YMMV.

Reply:
We need a way to pass scripts to concat that are just included as-is instead of wrapping in evals and I believe requirejs needs to be included before any of the manifest files and the require([<entryModule>]) needs to be afterwards which makes need for both pre and post scripts.

Reply:
This API could get improved with a better manifest file that could support eval/no-eval, entry module, code splitting, etc.. @alexeagle and I have been discussing this for code splitting